syntax = "proto2";

import "protos/model.proto";
import "protos/hyperparams.proto";


message AdViSEGCNModel {
  extend Model {
    optional AdViSEGCNModel ext = 1453;
  }

  // MLP hidden layers.
  optional int32 hidden_layers = 1 [deprecated = true];

  // MLP hidden units.
  optional int32 hidden_units = 2 [default = 200];

  // MLP output units.
  optional int32 output_units = 3 [default = 200];

  // Hyper-parameters of fully-connected layer.
  optional Hyperparams fc_hyperparams = 4;

  // Path to the GloVe file.
  optional string glove_path = 5;

  // Slogan vocab file.
  optional string slgn_vocab_list_path = 6;

  // Statement vocab file.
  optional string stmt_vocab_list_path = 7;

  // Margin of triplet loss.
  optional float triplet_margin = 8 [default = 0.2];

  // Dropout keep probability of the multi-modal dot-product operation.
  optional float dot_product_dropout_keep_prob = 9 [default = 0.5];

  optional int32 embedding_dims = 10 [default = 200];

  optional int32 inference_dims = 20 [default = 20];

  optional float max_norm = 12 [default = 6.0];

  optional bool add_image_as_a_roi = 11 [default = true];

  enum AdjacencyMatrix {
    IDENTITY = 1;
    DENSE = 2;
    SIMILARITY = 3;
  }
  optional AdjacencyMatrix stmt_adjacency = 33 [default = IDENTITY];
}
